addons/website_theme_install/models/ir_module_module.py

```diff
@@ -252,11 +318,13 @@ def button_choose_theme(self):
     def button_remove_theme(self):
         website = self.env['website'].get_current_website()
         self._remove_theme_on_website(website)
-        website.theme_id = False
 
     def button_refresh_theme(self):
         website = self.env['website'].get_current_website()
-        self._load_one_theme_module(website, with_update=True)
+
+        # we only need to update the common ancestor (theme_common) to update them all
```

Comment:
This assumes the automated upgrade work and is taking "old" into account correctly. Otherwise revert.